shallow_backup.py

```diff
@@ -365,7 +369,11 @@ def backup_prefix(path):
 		return os.path.join(configs_path, path)
 
 	configs_dir_mapping = {"Library/Application Support/Sublime Text 2/Packages/User/": "sublime_2",
-						   "Library/Application Support/Sublime Text 3/Packages/User/": "sublime_3", }
+						   "Library/Application Support/Sublime Text 3/Packages/User/": "sublime_3",
+						   "Library/Preferences/IntelliJIdea2018.2/":"intellijidea_2018-2",
+						   "Library/Preferences/PyCharm2018.2/":"pycharm_2018-2",
```

Comment:
One issue that jumps out to me here is that this only backs up specific versions of each of the IDEs. The way this should work is that it backs up the settings of the latest version of each editor. One solution is adding this mapping to the config JSON file, making it user-editable. Another (non-mutually exclusive) solution is to do something with wildcards like grabbing all the `Library/Preferences/IntelliJ*` paths and seeing which suffix is the largest/latest and backing that up. Then we could do some clever mapping stuff in the backed to have the only thing displayed in the config be something like:

```json
configs_for_backup: [
      "atom",
      "sublime",
      "intellij-idea",
      "intellij-ce,
      ...
]
```
We'd have to do some minor refactoring, but I think this would be a great UX improvement/feature.